python_lib/qss_compiler/compile.py

```diff
@@ -297,6 +297,16 @@ def _do_compile(
                 return_diagnostics=return_diagnostics,
             )
 
+        # Place all higher-level diagnostics related to user input here
+        # TODO: Best way to deal with multiple diagnostics?
+        for diag in diagnostics:
+            if diag.category == ErrorCategory.QSSCompilerSequenceTooLong:
+                raise exceptions.QSSCompilerSequenceTooLong(
+                    diag.message,  # TODO: Code reviewers: Is this safe?
+                    diagnostics,
+                    return_diagnostics=return_diagnostics,
+                )
```

Comment:
Oh.. I see what you're trying to do here. I think it should be safe. @bcdonovan Any thoughts?

Reply:
Yeah, my main question is about whether passing this message to the higher level is the correct thing to do. It seems like it probably is because we have the map that says whether or not it's safe to pass on to the user (no secret info in the message). But I just wanted to be sure people thought about it.

Reply:
Another alternative is to have specific messages (e.g. `Sequence too long`) or non-specific (`User input error`) and hopefully the more specific message would still get through in the case that the map says it's okay to pass on.

Reply:
Yeah I like the alternative too